source/Halibut/Transport/SecureConnection.cs

```diff
@@ -30,8 +40,10 @@ public void NotifyUsed()
 
         public bool HasExpired()
         {
-            // TODO - ASYNC ME UP!
-            // Use the HalibutRuntimeLimits
+            if (asyncHalibutFeature.IsEnabled())
+            {
+                return lastUsed < DateTimeOffset.UtcNow.Subtract(halibutTimeoutsAndLimits.SafeTcpClientPooledConnectionTimeout);
```

Comment:
We can keep them separate, but if we made this method simply be `return lastUsed < DateTimeOffset.UtcNow.Subtract(halibutTimeoutsAndLimits.SafeTcpClientPooledConnectionTimeout);` (without the `IsEnabled` check), would that not be the same?

Mainly because the default limits (which is what we use when we don't give any) use all the values from `HalibutLimits`, right?

Reply:
amazingly it will be null in that case.